jpo-ode-svcs/src/main/java/us/dot/its/jpo/ode/services/asn1/Asn1DecodedDataRouter.java

```diff
@@ -103,6 +103,16 @@ public Object process(String consumedData) {
 				// Send all TIMs also to OdeTimJson
 				timProducer.send(odeProperties.getKafkaTopicOdeTimJson(), getRecord().key(), odeTimData);
 				logger.debug("Submitted to TIM Pojo topic");
+				// If the TIM was generated by the TMC broadcast to the filtered TMC TIM Kafka topic
+				try {
+					String generatedBy = (new JSONObject(odeTimData)).getJSONObject("metadata")
+							.getString("recordGeneratedBy");
+					if (generatedBy.equalsIgnoreCase("TMC_VIA_SAT") || generatedBy.equalsIgnoreCase("TMC_VIA_SNMP")) {
+						timProducer.send(odeProperties.getKafkaTopicOdeTimJsonTMCFiltered(), null, odeTimData);
```

Comment:
Is it possible that we'll get duplicates in this new topic?

Reply:
This has been updated to send TMC-generated messages from one place instead of two so there won't be duplicate messages in the new topic